### PR TITLE
OP-949 fix classcastexception during Recity

### PR DESCRIPTION
**Pull request by dbmalkovsky**
See OP-949

I don't like the fix but it works.    Depending on whether or not the drug has any stock and/or whether the actual number is changed most of the time the value is a DOUBLE but in some paths it is an INTEGER.  The "fix" is to handle both cases.

If this is acceptable then I will create the hotfix PR.

### CODE REVIEW

**Review by mwithi — COMMENTED** [review]
It looked strange, so I've investigated a bit. According with the javadoc ([here](https://docs.oracle.com/javase/7/docs/api/javax/swing/SpinnerNumberModel.html)) relying on the model instead of the component should do the trick:
```
SpinnerNumberModel model = new SpinnerNumberModel(0.0d, 0.0d, null, 0.5d);
jSpinnerNewQty = new JSpinner(model);
jSpinnerNewQty.setFont(new Font("Tahoma", Font.BOLD, 14)); //$NON-NLS-1$
jSpinnerNewQty.addChangeListener(changeEvent -> {
	Double stock = Double.parseDouble(jLabelStockQty.getText());
	Double newQty = (Double) model.getNumber().doubleValue();
```

**Comment by dbmalkovsky** [comment]
OK; made your changes and a few more that I couldn't resist given I was in the file :)

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java [hunk 1]
```diff
@@ -366,19 +372,15 @@ private void initComponents() {
 			} catch (OHServiceException e2) {
 				OHServiceExceptionUtil.showMessages(e2);
 			}
-			double newQty = (Double) jSpinnerNewQty.getValue();
+			double newQty = (Double)spinnerNewQtyModel.getNumber().doubleValue();
```
Comment (mwithi) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:375]:
```suggestion
			double newQty = (Double) spinnerNewQtyModel.getNumber().doubleValue();
```

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java [hunk 2]
```diff
@@ -669,12 +672,15 @@ private JSpinner getJSpinnerNewQty() {
...
 			jLabelStockQty.setHorizontalAlignment(SwingConstants.CENTER);
 			jLabelStockQty.setPreferredSize(new Dimension(100, 25));
-			jLabelStockQty.setFont(new Font("Tahoma", Font.BOLD, 14)); //$NON-NLS-1$
+			jLabelStockQty.setFont(FONT_BOLD);
+			jLabelStockQty.setForeground(Color.BLACK);
+			jLabelStockQty.setEditable(false);
+			jLabelStockQty.setBorder(BorderFactory.createEmptyBorder());
 		}
 		return jLabelStockQty;
 	}
```
Comment (mwithi) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:686]: should the variable also be renamed to jTextFieldStockQty? 

Anyway I was suggesting to not "parse" its text, rather get it directly from the source variable

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java (hunk 1 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java]: OK; added a space

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:686]: OK; made the name change.   Sorry I didn't understand your original comment; but given that it is a text field this just feels better.

**Review by mwithi — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**